plugins/woocommerce/src/Internal/ComingSoon/ComingSoonHelper.php

```diff
@@ -0,0 +1,71 @@
+<?php
+namespace Automattic\WooCommerce\Internal\ComingSoon;
+
+use Automattic\WooCommerce\Admin\WCAdminHelper;
+
+/**
+ * Provides helper methods for coming soon functionality.
+ */
+class ComingSoonHelper {
+
+	/**
+	 * Returns true when the entire site is live.
+	 */
+	public function is_site_live(): bool {
+		return 'yes' !== get_option( 'woocommerce_coming_soon' );
+	}
+
+	/**
+	 * Returns true when the entire site is coming soon mode.
+	 */
+	public function is_site_coming_soon(): bool {
+		return 'yes' === get_option( 'woocommerce_coming_soon' ) && 'yes' !== get_option( 'woocommerce_store_pages_only' );
+	}
+
+	/**
+	 * Returns true when only the store pages are in coming soon mode.
+	 */
+	public function is_store_coming_soon(): bool {
+		return 'yes' === get_option( 'woocommerce_coming_soon' ) && 'yes' === get_option( 'woocommerce_store_pages_only' );
+	}
+
+	/**
+	 * Returns true when the provided URL is behind a coming soon screen.
+	 *
+	 * @param string $url The URL to check.
+	 */
+	public function is_url_coming_soon( string $url ): bool {
+		// Early exit if coming soon mode not active.
+		if ( $this->is_site_live() ) {
+			return false;
+		}
+
+		if ( $this->is_site_coming_soon() ) {
+			return true;
+		}
+
+		// Check the URL is a store page when in "store coming soon" mode.
+		if ( $this->is_store_coming_soon() && WCAdminHelper::is_store_page( $url ) ) {
+			return true;
+		}
+
+		// Default to false.
+		return false;
+	}
+
+	/**
+	 * Builds the relative URL from the WP instance.
+	 *
+	 * @internal
+	 * @link https://wordpress.stackexchange.com/a/274572
+	 * @param \WP $wp WordPress environment instance.
+	 */
+	public function get_url_from_wp( \WP $wp ) {
+		// Special case for plain permalinks.
+		if ( empty( get_option( 'permalink_structure' ) ) ) {
+			return '/' . add_query_arg( $wp->query_vars, $wp->request );
```

Comment:
This is pretty foreign to me but should we do some sanitisation here?

Reply:
I think you might be right. I was assuming it was already sanitized, but [this](https://github.com/WordPress/wordpress-develop/blob/5afdac7e057bf4faf097079c0704f6e834fdce97/src/wp-includes/class-wp.php#L330-L333) indicates raw values are being copied over. If that's the case, there would be a risk of an attacker crafting a URL param to trick the site into showing live content 🤔  I'll investigate more in a follow-up issue.